features/support/commonlib.rb

```diff
@@ -3,34 +3,25 @@
 
 require 'tempfile'
 
-# Class for generating expect files to interactively run spacewalk-push-register.
-class ExpectFileGenerator
-  def initialize(host, bootstrap)
-    raise(ArgumentError, 'Hostname is missing!') if host.nil? || host.empty?
-    Tempfile.open('push-registration.expect') do |f|
-      @file = f
-      f.write('spawn spacewalk-ssh-push-init --client ' + host + ' --register ' + bootstrap + ' --tunnel' + "\n")
-      f.write("while {1} {\n")
-      f.write("  expect {\n")
-      f.write("    eof                                                        {break}\n")
-      f.write("    \"Are you sure you want to continue connecting (yes/no)?\" {send \"yes\r\"}\n")
-      f.write("    \"Password:\"                                              {send \"linux\r\"}\n")
-      f.write("  }\n")
-      f.write("}\n")
-    end
-  end
-
-  # Return the file path.
-  def path
-    @file.path
-  end
-
-  # Return the file name.
-  def filename
-    File.basename(path)
+# generate expect file to interactively run spacewalk-push-register
+def generate_expect_file(host, bootstrap)
+  raise(ArgumentError, 'Hostname is missing!') if host.nil? || host.empty?
+  path = ''
+  Tempfile.open('push-registration.expect') do |f|
+    path = f.path
+    f.write('spawn spacewalk-ssh-push-init --client ' + host + ' --register ' + bootstrap + ' --tunnel' + "\n")
+    f.write("while {1} {\n")
+    f.write("  expect {\n")
```

Comment:
Create a constant Var like TEXT = "blabla".freeze, note the freeze in ruby, so we remove data that changes from code logic and we have only 1 time file writing